src/map/ai/helpers/pathfind.cpp

```diff
@@ -588,6 +590,42 @@ float CPathFind::GetRealSpeed()
 
             realSpeed *= 1.0f + speedMod / 100.0f;
         }
+
+        if (settings::get<bool>("map.USE_MOB_SPEED_BOOST_MULTIPLIER"))
+        {
+            if (auto* mobEntity = dynamic_cast<CMobEntity*>(m_POwner))
+            {
+                bool mobIsPlayerPet = mobEntity->PMaster && mobEntity->PMaster->objtype == TYPE_PC;
+
+                auto* battleTarget         = mobEntity->GetBattleTarget();
+                bool  targetWouldBeDrawnin = battleTarget && (distance(mobEntity->loc.p, battleTarget->loc.p) - battleTarget->m_ModelRadius) > mobEntity->GetMeleeRange();
+                bool  isEngaged            = mobEntity->animation == ANIMATION_ATTACK;
+
+                if (isEngaged && !mobIsPlayerPet && targetWouldBeDrawnin)
+                {
+                    // use default multiplier for most mobs
+                    float multiplier = settings::get<float>("map.DEFAULT_MOB_SPEED_BOOST_MULTIPLIER");
+
+                    // unless the mob has a custom multiplier
+                    if (mobEntity->getMobMod(MOBMOD_SPEED_BOOST_MULT) > 0)
+                    {
+                        multiplier = mobEntity->getMobMod(MOBMOD_SPEED_BOOST_MULT);
+                    }
+
+                    // clamp the multiplier between 1 and 5 to prevent underflows and overflows
+                    multiplier = std::clamp<float>(multiplier, 1.0f, 5.0f);
```

Comment:
> Probably clamp this after the weight penalty adjustment below, since if it's lower than 2.0 it'll actually decrease speed in chase mode.

This should now be fixed.